<launch>

  <arg name="image_view" default="true" />

  <!-- <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
    <param name="video_device" value="/dev/video0" />

    <param name="intrinsic_controls/focus_automatic_continuous" value="false" />
    <param name="image_width" value="1920" />
    <param name="image_height" value="1080" />
    <param name="pixel_format" value="yuv" />
    <param name="color_format" value="yuv422p" />
    <param name="camera_frame_id" value="usb_cam" />
    <param name="io_method" value="mmap"/>
    <param name="camera_info_url" value="file://$(find hmip_framework)/config/calibration/usb_camera_calibration_1920.yaml"/>    
  </node> -->

  <!-- TRACKERS -->
  <node name="hand_tracker" pkg="hand_gaze_trackers" type="handtrack.py" output="screen" >
  </node>

  <node name="gaze_tracker" pkg="hand_gaze_trackers" type="gazetrack.py" output="screen" >
  </node>

  <!-- PREDICTORS -->
  <node name="predict_hand_vf" pkg="hmip_framework" type="predict_hand_vf.py" output="screen" >
     <rosparam command="load" file="$(find hmip_framework)/config/object_locations_dataset01.yaml" />
     <param name="beta" value="1"/>
     <param name="disc_param" value="0.05"/>
  </node>

  <node name="predict_gaze" pkg="hmip_framework" type="predict_gaze.py" output="screen" >
      <rosparam command="load" file="$(find hmip_framework)/config/object_locations_dataset01.yaml" />
      <param name="gaze_hit_threshold" value="0.985"/>
  </node>

  <!-- ATTENTION PERCEPTION-->
  <node name="predict_nbc" pkg="hmip_framework" type="predict_nbc.py" output="screen" >
     <rosparam command="load" file="$(find hmip_framework)/config/object_locations_dataset01.yaml" />
  </node>


 <node if="$(arg image_view)" name="gaze_image_view" pkg="image_view" type="image_view"
        respawn="false" output="screen">
    <remap from="image" to="/gaze_tracker/image"/>
    <param name="autosize" value="true" />
  </node>

  <node if="$(arg image_view)" name="hand_image_view" pkg="image_view" type="image_view"
        respawn="false" output="screen">
    <remap from="image" to="/hand_tracker/image"/>
    <param name="autosize" value="true" />
  </node>

</launch>

